velox/exec/tests/WindowTest.cpp

```diff
@@ -79,6 +81,262 @@ TEST_F(WindowTest, spill) {
+            // Sorting key.
+            makeFlatVector<int32_t>(size, [](auto row) { return row; }),
+        });
+
+    createDuckDbTable({data});
+
+    // Abstract the common values vector split.
+    auto valuesSplit = split(data, 10);
+    core::PlanNodeId windowId;
+    auto builder = PlanBuilder().values(valuesSplit);
+    if (useStreamingWindow) {
+      builder.orderBy({"p", "s"}, false)
+          .streamingWindow({"row_number() over (partition by p order by s)"});
+    } else {
+      builder.window({"row_number() over (partition by p order by s)"});
+    }
+    auto plan = builder.capturePlanNodeId(windowId).planNode();
+    auto task =
+        AssertQueryBuilder(plan, duckDbQueryRunner_)
+            .config(core::QueryConfig::kPreferredOutputBatchBytes, "1024")
+            .assertResults(
+                "SELECT *, row_number() over (partition by p order by s) FROM tmp");
+
+    return exec::toPlanStats(task->taskStats()).at(windowId).peakMemoryBytes;
+  };
+
+  const vector_size_t smallSize = 100'000;
+  const vector_size_t largeSize = 1'000'000;
+  // As the volume of data increases, the peak memory usage of the sort-based
+  // window will increase (2418624 vs 17098688). Since the peak memory usage of
+  // the RowBased Window represents the one batch data in a single partition,
+  // the peak memory usage will not increase as the volume of data grows.
+  auto sortWindowSmallUsage = memoryUsage(false, smallSize);
+  auto sortWindowLargeUsage = memoryUsage(false, largeSize);
+  ASSERT_GT(sortWindowLargeUsage, sortWindowSmallUsage);
+
+  auto rowWindowSmallUsage = memoryUsage(true, smallSize);
+  auto rowWindowLargeUsage = memoryUsage(true, largeSize);
+  ASSERT_EQ(rowWindowSmallUsage, rowWindowLargeUsage);
+}
+
+TEST_F(WindowTest, rankRowStreamingWindowBuild) {
+  auto data = makeRowVector(
+      {"c1"},
+      {makeFlatVector<int64_t>(std::vector<int64_t>{1, 1, 1, 1, 1, 2, 2})});
+
+  createDuckDbTable({data});
+
+  const std::vector<std::string> kClauses = {
+      "rank() over (order by c1 rows unbounded preceding)"};
+
+  auto plan = PlanBuilder()
+                  .values({data})
+                  .orderBy({"c1"}, false)
+                  .streamingWindow(kClauses)
+                  .planNode();
+
+  std::atomic_bool isStreamCreated{false};
+  SCOPED_TESTVALUE_SET(
+      "facebook::velox::exec::RowsStreamingWindowBuild::RowsStreamingWindowBuild",
+      std::function<void(RowsStreamingWindowBuild*)>(
+          [&](RowsStreamingWindowBuild* windowBuild) {
+            isStreamCreated.store(true);
+          }));
+
+  AssertQueryBuilder(plan, duckDbQueryRunner_)
+      .config(core::QueryConfig::kPreferredOutputBatchBytes, "1024")
+      .config(core::QueryConfig::kPreferredOutputBatchRows, "2")
+      .config(core::QueryConfig::kMaxOutputBatchRows, "2")
+      .assertResults(
+          "SELECT *, rank() over (order by c1 rows unbounded preceding) FROM tmp");
+
+  ASSERT_TRUE(isStreamCreated.load());
```

Comment:
@xiaoxmeng This ut failed in ci with following exception. But can pass in my local env. Do you have any input? Thanks.

```
2024-08-19T09:11:33.2777347Z [ RUN      ] WindowTest.rankRowStreamingWindowBuild
2024-08-19T09:11:33.2777548Z /__w/velox/velox/velox/exec/tests/WindowTest.cpp:225: Failure
2024-08-19T09:11:33.2777661Z Value of: isStreamCreated.load()
2024-08-19T09:11:33.2777744Z   Actual: false
2024-08-19T09:11:33.2777828Z Expected: true
```

Reply:
Use DEBUG_ONLY_TEST_F for test cases using testvalue as it only works on debug build.